src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicTlsSecret.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Quic;
+using static Microsoft.Quic.MsQuic;
+
+namespace System.Net.Quic;
+
+internal sealed class MsQuicTlsSecret : IDisposable
```

Comment:
Should we make the whole class `DEBUG` only? I.e. non-existent in Release build.

Reply:
I was originally thinking I would make it conditional in project file but we suck in all files automatically. I assumed it would be trimmed away but wrapping it in `#if debug` is probably better idea.

Reply:
Yeah, that's true, we have wildcard in project file. I was more worried that somebody might try to use it elsewhere, or copy parts of it. And then it would end up in production somewhere. The added `#if DEBUG` would express the intent clearly, but I just might be overthinking it.

Reply:
I do like the proposal. This is start and we may work out process for users to get hands on the capability.